sklearn/impute.py

```diff
@@ -40,6 +40,18 @@
 ]
 
 
+def _check_inputs_dtype(X, missing_values):
+    """Check that the type of missing_values is in accordance with the
+    dtype of X.
+    """
+    if (X.dtype.kind in ("f", "i", "u") and
+            not isinstance(missing_values, numbers.Real)):
+        raise ValueError("The data type of 'missing_values' and 'X' are "
```

Comment:
Can you add what is expected? In general it's better when the error message says something like: "this is what was expected. Got this instead."